design-library/src/components/BccTabs/BccTabs.stories.ts

```diff
@@ -0,0 +1,158 @@
+import { AccountTreeIcon, CheckIcon, CircleNotificationsIcon } from "@bcc-code/icons-vue";
+import BccTabs from "./BccTabs.vue";
+
+import type { Meta, StoryFn } from "@storybook/vue3";
+import { h } from "vue";
+
+export default {
+  title: "Components/BccTabs",
+  component: BccTabs,
+  argTypes: {
+    tabs: {
+      description: "Array of tabs you want to display",
+    },
+    size: {
+      description: "The size of the tabs",
+      options: ["sm", "base", "lg"],
+      control: { type: "radio" },
+    },
+  },
+} as Meta<typeof BccTabs>;
+
+const Template: StoryFn<typeof BccTabs> = (args) => ({
+  components: { BccTabs },
+  setup() {
+    return { args };
+  },
+  template: `<BccTabs v-bind="args"></BccTabs>`,
+});
+
+export const Example = Template.bind({});
+Example.args = {
+  tabs: [
+    {
+      title: "Using slots",
+      icon: CircleNotificationsIcon,
+      as: h({
+        template: `<div class="p-4">
+          <p>Define tab content using slots or components:</p>
+          <code class="my-2 inline-block border-4 border-dashed border-gray-300 p-2 text-gray-600">
+            &lt;template #tab-1&gt;You content here&lt;/template&gt;
+          </code>
+        </div>`,
+      }),
```

Comment:
Nice usage of the `as` option 👍

[Docs] I was wondering about a use case for it since it's a slot, when would you normally use this? Could you perhaps show that in the public documentation as well?

Reply:
Hmm, guess it really depends. You can use both a mix of slots and `as` just on different tabs. The main benefit of `as` is so you have a cleaner template and not have to deal with trying to match up tab indexes to their position in the `tabs` array.